cmd/build.go

```diff
@@ -193,7 +195,24 @@ func (b *buildCmd) createBuildTask() (*graph.Task, error) {
 	// TODO: create secrets
 	secrets := []*graph.Secret{}
 
-	return graph.NewTask(steps, secrets, b.opts.Registry, b.registryUser, b.registryPw, taskTotalTimeoutInSec, true)
+	var credentials []*graph.Credential
+	// If the user provides the username and password, add it to the Credentials
+	if b.opts.Registry != "" && b.registryUser != "" && b.registryPw != "" {
+		creds, _ := graph.NewCredential(b.opts.Registry, b.registryUser, b.registryPw)
```

Comment:
Even though this can't ever error, just handle it anyways since validation of `NewCredential` could change in the future.